|| [1.Installation](readme.md) ||  [2.Getting Started](gettingstarted.md) || [***3.Output***](output.md) || [4.Application](uses.md) || 

--NEED TO UPDATE PICTURES WITH FINAL VERSION--

In this section, we walk through a sample output/outputs generated by the sample input data [here]. We will cover:

**What is in the output folder section**
  - What each file means
  - How are the graphs generated and scaled
  - How is the phylip file generated


## What is in the output folder
After running your data or sample data linked above, you will find this 
folder in your current directory  or the directory you specified:

![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo_of_the_folder.PNG)

The time stamp refers to the  time you started the run. Opening, the folder you will find 2 folder, outpt and temp. temp contains program intermidates, so can help us trace errors if anything broke and you submitted an issue. 
![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo2.PNG)

Output folder is where the program output is. Opening it, you can find 

Here you can see  *(refname)_output* which contain all the analysis for this reference.We will get to that soon.


Next thing you will find is ReadMe file.It will look something like this 
![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo1.PNG)

This read me tell you what indexes that you will see later when we go to the the specific refrences _output file means.  In short, we refer to your reads paired or unpaired in an index. This makes it easier to name the files and manage output. We convert it back to your read name when analyzing graphs and summary tables. This is just for the folders.  

One of the important outputs is the Run_summar.csv table. It contains very useful info about the refrence length you used, average coverage for reads, percent coverage, base_coverage,etc. It look like this and as you notice we have the reads for you allowing  easy scanning  of the table. 

Since The run had single copy genes for normalization, there will be some info about single copy genes at the bototm of the table

![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo10.PNG)

Each row represt the refrence and the read used, so if you had for example 4 pairs of reads/4 unpaired reads, you will have 4 rows with the same refrence name and each row will contain the corrsoponding attributes shown 


The folder called plots_single_copy contains coverage plots for each single copy gene inputed. This helps in vizualizing the abudance of the single_copy genes provided for normalization

The folder called map_depth_allrefs contains depth per postion for every sample per reference. 



Lets say we will go the first folder which is refrence named Ref1_DmelR1_output.  This reference is D.mel R1 element 

You will find  the folder look like this 

![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo3.PNG)

First thing you will notice is these sub folders. Each subfolder belong to the sample (this can be either read or read pair) for this specific reference sequence. The indexes which we talked about earlier. We will go into the subfolders later.

Now, next thing you will notice  is *scaled_profiles.pdf* file . 

![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/phtoto6.PNG)
=
Each of these graph represent the coverage of the read/read pair when mapped to that  specfic refrence we opened its folder.  They are all based on a single color scale. That means by just quick looking at the graph you can tell which parts of the refrence sequence (refrence sequence base position is on x axis ) has the highest (red)/lowest(blue)  coverage. This  can tell you a lot about this read and refrence sequence. How/where this sequence enriched in that species reads compared to the other species reads. Color scaling accross different reads will help you in a great way to vizualize differences  among very large group of data.


Next pdf in the output folder is the *variant_profiles.pdf*
These plots has the variation(mismatches) in  among all the reads for this specific refrences. They have been put together for your convience to compare. 


![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo7.PNG)


For example, by investigating this pdf, user can tell which regions of the reference has common mismatches between samples 


Opening one of the sub folders will look like this
![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo5.PNG)

it contain  the same plots we looked at earlier but indvidully. Keep in mind they are still scaled among all reads, so its main use if you wanted to use a high quality indvidual graph.A text file that contains the information per every position in the refrence sequence. It contains the depth of every postion/mismatches of A/mismatches of T/mismatches of C//mismatches of G/. 





Now  lets go back and check what is in the Phylip file and what does it mean  

![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo8.PNG)


Based on the variation plots explained earlier, we were able to capture phylogentic informative sites from these plots. We analysed each plot indivudaly and recored the bases that had high mis match coverage over 10% from the actual coverage. We also accounted for ambigous sites.

 This phylip files are most informative for long refrence sequences (1000+) bases. You can run this phylip file in your favorite tree software. We use iq-tree. Check the detalied tutorial on the website for on how to choose which phylip files to get most accurate relationships between the samples  *upcoming*
 

 
 Back to  main output folder, You will find a table called
 *References_summary_base_coverage*.
 ![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo1A.PNG)
 
This tells you the mean base coverage between all samples for a reference (max value is 1). It is helpful you can discard references with low mean base coverage from phylogentic analysis as their phylip files will not be very imformative 

The folder called *scaled_profiles_allrefs* contains coverage plots scaled based on entire run not just a single reference


#####  Correlation analysis output

4 output folders are outputed in Correlation analysis.

*First folder*: correlation_boxplots_by_group
This folder contains boxplots for each group you inputed in your user_groups.txt. The boxplots show how does a group vary among all references used in the run. Each group has its own plot.

*Second folder*: correlation_boxplots_by_reference
 This folder contains boxplots that compare profile shape within and between user-defined groups, organized by reference.

*Third folder*: correlation_data
 This folder contains correlation matrices for each reference. The matrices shows all the correlation values among all samples for a reference. Each refrence has its own matrix.

*Fourth folder*:correlation_histograms
 This folder contains histograms of within, and between group correlation values for each reference.

*Last file*: correlation_summary.csv
  This table contains average values of within and between each group correlation value for each reference. It summarizes information for all correlation done in the run

![](https://github.com/johnssproul/RepeatProfiler/raw/master/pics_readme/photo2A.png)
